pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -1360,3 +1365,22 @@ func bindVarsFn(fn func(int) string, query string) string {
 	buf.WriteString(query)
 	return buf.String()
 }
+
+func validateDBConfig(cfg *configuration) error {
+	if cfg.DatabaseType == "" {
+		return errors.New("database_type must be set")
+	}
+
+	if cfg.ConnectionString == "" {
+		return errors.New("connection_string must be set")
+	}
+
+	switch cfg.DatabaseType {
```

Comment:
nit: we don't need this switch statement if each dialect takes care of validating the connection string inside of the various connect() methods.